internal/pkg/cli/app_show.go

```diff
@@ -30,36 +29,42 @@ const (
 	applicationShowAppNameHelpPrompt     = "The detail of an application will be shown (e.g., endpoint URL, CPU, Memory)."
 )
 
-type serializedWebAppEnv struct {
-	Name   string `json:"name"`
-	Region string `json:"region"`
-	Prod   bool   `json:"prod"`
-	URL    string `json:"url"`
-	Path   string `json:"path"`
+type serializedWebAppConfig struct {
+	Environment string `json:"environment"`
+	Port        string `json:"port"`
+	Tasks       string `json:"tasks"`
+	CPU         string `json:"cpu"`
+	Memory      string `json:"memory"`
 }
 
-type serializedWebAppService struct {
-	Port   string `json:"port"`
-	Tasks  string `json:"tasks"`
-	CPU    string `json:"cpu"`
-	Memory string `json:"memory"`
+type serializedWebAppRoute struct {
+	Environment string `json:"environment"`
+	URL         string `json:"url"`
+	Path        string `json:"path"`
+}
+
+type serializedWebAppInfra struct {
+	Environment      string `json:"environment"`
+	ServiceID        string `json:"serviceID"`
+	TargetGroupID    string `json:"targetGroupID"`
+	TaskDefinitionID string `json:"taskDefinitionID"`
 }
 
 type serializedWebApp struct {
-	AppName      string                    `json:"appName"`
-	Type         string                    `json:"type"`
-	Project      string                    `json:"project"`
-	Account      string                    `json:"account"`
-	Environments []serializedWebAppEnv     `json:"environments"`
-	Services     []serializedWebAppService `json:"services"`
+	AppName         string                   `json:"appName"`
+	Type            string                   `json:"type"`
+	Project         string                   `json:"project"`
+	Configurations  []serializedWebAppConfig `json:"configurations"`
+	Routes          []serializedWebAppRoute  `json:"routes"`
+	Infrastructures []serializedWebAppInfra  `json:"infrastructures,omitempty"`
 }
 
 // ShowAppOpts contains the fields to collect for showing an application.
 type ShowAppOpts struct {
-	ShouldOutputJSON bool
+	shouldOutputJSON  bool
+	shouldOutputInfra bool
 
-	proj *archer.Project
-	app  serializedWebApp
+	app serializedWebApp
```

Comment:
nit: is there a benefit of storing this in the state? instead of returning it in `retrieveData`?